drivers/metal/SCsub

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env python
+
+Import("env")
+
+env_metal = env.Clone()
+if "-fno-exceptions" in env_metal["CXXFLAGS"]:
+    env_metal["CXXFLAGS"].remove("-fno-exceptions")
+if "-std=gnu++17" in env_metal["CXXFLAGS"]:
+    env_metal["CXXFLAGS"].remove("-std=gnu++17")
+env_metal.Append(CXXFLAGS=["-fexceptions", "-std=c++20"])
```

Comment:
So I guess Metal requires exceptions and C++20?
Or is it that you chose to use both in your own code?

We might consider using C++20 in the near future but for now I'd prefer avoiding mixing standards if we can. I'm also not sure how well compilers deal with this when doing optimization, especially LTO on the whole project.

If the implementation would be significantly more convoluted with C++17, I'm open to accepting this as is though, but I wanted to raise it for discussion.

Likewise for exceptions, Godot doesn't use exceptions by design, and instead does its best to recover gracefully from errors in non game-breaking ways. It's fine for software developers to run into exceptions, but not for gamers, who are the consumers of our software.

Reply:
> So I guess Metal requires exceptions 

I'm forced to use exceptions due to `spirv-cross`, which either has exceptions or assertions:

https://github.com/KhronosGroup/SPIRV-Cross/blob/66363ac7e8fe4465169ceb2ad5905b1657244cfe/spirv_cross_error_handling.hpp#L30

Assertions are a bad, given they terminate the process. We handle the exceptions and use regular Godot error practices if spriv-cross APIs fails. We'd have to make significant changes to spriv-cross to remove exceptions, which I don't think we would want to maintain.

> and C++20?

I'm only using that directive in the Objective-C++ code files, which are already a mix of C++ and Objective-C. C++20 was used for making the serialisation code simpler by using, but it is not exposed to any other parts of Godot.